snuba/datasets/entities/factory.py

```diff
@@ -0,0 +1,57 @@
+from typing import Callable, MutableMapping, Set, Sequence
+
+from snuba import settings
+from snuba.datasets.entity import Entity
+from snuba.util import with_span
+
+ENTITY_IMPL: MutableMapping[str, Entity] = {}
+ENTITY_NAME_LOOKUP: MutableMapping[Entity, str] = {}
+
+ENTITY_NAMES: Set[str] = {
+    "outcomes",
+    "outcomes_raw",
+}
+
+ACTIVE_ENTITY_NAMES: Set[str] = ENTITY_NAMES - settings.DISABLED_ENTITIES
+
+
+class InvalidEntityError(Exception):
+    """Exception raised on invalid entity access."""
+
+
+@with_span()
+def get_entity(name: str) -> Entity:
+    if name in ENTITY_IMPL:
+        return ENTITY_IMPL[name]
+
+    if name in settings.DISABLED_ENTITIES:
+        raise InvalidEntityError(
+            f"entity {name!r} is not available in this environment"
+        )
+
+    from snuba.datasets.entities.outcomes import OutcomesEntity
+    from snuba.datasets.entities.outcomes_raw import OutcomesRawEntity
+
+    entity_factories: MutableMapping[str, Callable[[], Entity]] = {
+        "outcomes": OutcomesEntity,
+        "outcomes_raw": OutcomesRawEntity,
+    }
+
+    try:
+        entity = ENTITY_IMPL[name] = entity_factories[name]()
+        ENTITY_NAME_LOOKUP[entity] = name
+    except KeyError as error:
+        raise InvalidEntityError(f"entity {name!r} does not exist") from error
+
+    return entity
+
+
+def get_entity_name(entity: Entity) -> str:
+    try:
+        return ENTITY_NAME_LOOKUP[entity]
+    except KeyError as error:
+        raise InvalidEntityError("Entity name not specified") from error
+
+
+def get_enabled_dataset_names() -> Sequence[str]:
```

Comment:
Should this be `get_enabled_entity_names()`? And, should this be at a dataset level or do we plan to enable an entity name across the board?

Reply:
Typo, my mistake. As for whether entities also need to be enabled/disabled, I need to dig more into what that means.